src/__tests__/parser.test.ts

```diff
@@ -0,0 +1,457 @@
+      # Test file
+      
+      # description: Your Twilio Account SID
+      # format: sid
+      TWILIO_ACCOUNT_SID=ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      
+      # description: Your Twilio Auth Token
+      # format: secret
+      TWILIO_AUTH_TOKEN=
+    `;
+      const result = parse(file);
+      expect(result.variables).toEqual([
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_ACCOUNT_SID',
+          description: 'Your Twilio Account SID',
+          format: 'sid',
+          default: 'ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
+        },
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_AUTH_TOKEN',
+          description: 'Your Twilio Auth Token',
+          format: 'secret',
+          default: null,
+        },
+      ]);
+    });
+
+    test('handles invalid format values', () => {
+      const file = stripIndent`
+      # Test file
+      
+      # description: Your Twilio Account SID
+      # format: boolean
+      TWILIO_ACCOUNT_SID=ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      
+      # description: Your Twilio Auth Token
+      # format: secret
+      TWILIO_AUTH_TOKEN=
+    `;
+
+      expect(() => parse(file)).toThrowError();
+    });
+
+    test('handles invalid list format values', () => {
+      const file = stripIndent`
+      # Test file
+      
+      # description: Your Twilio Account SID
+      # format: list(boolean)
+      TWILIO_ACCOUNT_SID=ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      
+      # description: Your Twilio Auth Token
+      # format: secret
+      TWILIO_AUTH_TOKEN=
+    `;
+
+      expect(() => parse(file)).toThrowError();
+    });
+
+    test('handles invalid nested_list values', () => {
+      const file = stripIndent`
+      # Test file
+      
+      # description: Your Twilio Account SID
+      # format: nest_list(phone_number,boolean)
+      TWILIO_ACCOUNT_SID=ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+      
+      # description: Your Twilio Auth Token
+      # format: secret
+      TWILIO_AUTH_TOKEN=
+    `;
+
+      expect(() => parse(file)).toThrowError();
+    });
+
+    test('recognizes required comment', () => {
+      const file = stripIndent`
+      # Test file
+      
+      # description: your Twilio phone number
+      # format: phone_number
+      # required: true
+      TWILIO_PHONE_NUMBER=+12223334444
+
+      # description: Message to play
+      # required: false
+      MESSAGE=
+    `;
+      const result = parse(file);
+      expect(result.variables).toEqual([
+        {
+          ...DEFAULT_ENTRY,
+          key: 'TWILIO_PHONE_NUMBER',
+          format: 'phone_number',
+          required: true,
+          description: 'your Twilio phone number',
+          default: '+12223334444',
+        },
+        {
+          ...DEFAULT_ENTRY,
+          key: 'MESSAGE',
+          required: false,
+          description: 'Message to play',
+          default: null,
+        },
+      ]);
+    });
+
+    test('recognizes link comment', () => {
```

Comment:
is there a possible invalid case to test?

Reply:
I think we had decided to not do validation on the link which is why I left it raw. But I could do some parsing on it to validate it, which would introduce invalid cases to test